src/ol/PluggableMap.js

```diff
@@ -1424,9 +1433,12 @@ function getLoading(layers) {
     if (layer instanceof LayerGroup) {
       return getLoading(layer.getLayers().getArray());
     }
-    const source = layers[i].getSource();
-    if (source && source.loading) {
-      return true;
+
+    if (layer instanceof Layer) {
```

Comment:
Can simply be an `else` block, if necessary with a type cast to `Layer`. We should try to avoid `instanceof` checks wherever possible.